File: library/src/main/kotlin/tag/type/application/ApplicationData.kt
package tag.type.application

data class ApplicationData(
    val type: UInt,
    val version: UInt,
    val capabilities: UInt,
    val productId: UByte
) {
    companion object {
        const val APP_TYPE: UInt = 32U
        const val APP_VERSION: UInt = 5U
        const val APP_CAPABILITIES: UInt = 0U
        const val APP_PRODUCT_ID: UByte = 54U
    }
}

File: python-library/tag/type/application/application_data.py
import struct
from dataclasses import dataclass


@dataclass
class ApplicationData:
    """Application tag data structure"""

    type: int
    version: int
    capabilities: int
    product_id: int

    # Constants як у Kotlin companion object
    APP_TYPE: int = 32
    APP_VERSION: int = 5
    APP_CAPABILITIES: int = 0
    APP_PRODUCT_ID: int = 54

    def content(self) -> bytes:
        """
        Convert to bytes (little-endian)

        Returns:
            bytes: 13-byte array (4+4+4+1 bytes)
        """
        # Pack as little-endian: 3 x UInt (4 bytes) + 1 x UByte (1 byte)
        return struct.pack('<IIIB', self.type, self.version, self.capabilities, self.product_id)
